spec/tree_spec.rb

```diff
@@ -1,14 +1,80 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe AppleTree do
+  context 'when AppleTree exists' do
+    subject(:tree) { AppleTree.new }
+    it 'should be a Class' do
+      expect(described_class.is_a? Class).to be true
+    end
+
+    it 'should be a newborn tree' do
+      expect(tree.age).to eq 0
+    end
+
+    it 'should have no apples' do
+      expect(tree.any_apples?).to eq false
+    end
+
+    it 'should not be able to have an apple picked' do
+      expect{tree.pick_an_apple!}.to raise_error NoApplesError
+    end
+
+    it 'should be alive' do
+      expect(tree.alive).to eq true
+    end
+
+    it 'should age when told to' do
+      expect(tree.age!).to eq 1
+    end
+
+    it 'should get taller as it ages' do
+      tree.age!
+      expect(tree.height).to be > 0
+    end
+
+    it 'should die after 10 years' do
+      10.times {
+        tree.age!
+      }
+      expect(tree.dead?).to eq true
+    end
```

Comment:
Not that you need to add it, but think about testing the flip side here too: this only shows that tens years of aging kills the tree - but maybe 9 or 8 years would also kill it.